app/services/edition_link_extractor.rb

```diff
@@ -0,0 +1,43 @@
+require "govspeak/link_extractor"
+
+class EditionLinkExtractor
+  def initialize(edition:)
+    @edition = edition
+  end
+
+  def call
+    if has_parts?
+      links_in_govspeak_fields + links_in_parts
+    else
+      links_in_govspeak_fields
+    end
+  end
+
+private
+
+  attr_reader :edition
+
+  def has_parts?
+    edition.parts.any?
```

Comment:
Not too sure/sold on this but I couldn't get `edition.has_attribute?()` to return correctly

Reply:
Which I assume is because they are embedded?

Reply:
Is it because an edition doesn't have a `parts` or that `parts` is `nil` so `any?` doesn't exist?